.buildkite/scripts/steps/esql_grammar_sync.sh

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+synchronize_lexer_grammar () {
+  license_header="$1"
+  source_file="$PARENT_DIR/elasticsearch/x-pack/plugin/esql/src/main/antlr/EsqlBaseLexer.g4"
+  destination_file="./packages/kbn-monaco/src/esql/antlr/esql_lexer.g4"
+
+  # Copy the file
+  cp "$source_file" "$destination_file" 
+
+  # Insert the license header
+  temp_file=$(mktemp)
+  printf "%s\n\n// DO NOT MODIFY THIS FILE BY HAND. IT IS MANAGED BY A CI JOB.\n\n%s" "$license_header" "$(cat $destination_file)" > "$temp_file"
+  mv "$temp_file" "$destination_file"
+
+  # Replace the line containing "lexer grammar" with "lexer grammar esql_lexer;"
+  sed -i -e 's/lexer grammar.*$/lexer grammar esql_lexer;/' "$destination_file" 
+
+  # Insert "options { caseInsensitive = true; }" one line below
+  sed -i -e '/lexer grammar esql_lexer;/a\
+  options { caseInsensitive = true; }' "$destination_file" 
+
+  echo "File copied and modified successfully."
+}
+
+synchronize_parser_grammar () {
+  license_header="$1"
+  source_file="$PARENT_DIR/elasticsearch/x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4"
+  destination_file="./packages/kbn-monaco/src/esql/antlr/esql_parser.g4"
+
+  # Copy the file
+  cp "$source_file" "$destination_file" 
+
+  # Insert the license header
+  temp_file=$(mktemp)
+  printf "%s\n\n// DO NOT MODIFY THIS FILE BY HAND. IT IS MANAGED BY A CI JOB.\n\n%s" "$license_header" "$(cat ${destination_file})" > "$temp_file"
+  mv "$temp_file" "$destination_file"
+
+  # Replace the line containing "parser grammar" with "parser grammar esql_parser;"
+  sed -i -e 's/parser grammar.*$/parser grammar esql_parser;/' "$destination_file" 
+
+  # Replace options {tokenVocab=EsqlBaseLexer;} with options {tokenVocab=esql_lexer;}
+  sed -i -e 's/options {tokenVocab=EsqlBaseLexer;}/options {tokenVocab=esql_lexer;}/' "$destination_file" 
+
+  echo "File copied and modified successfully."
+}
+
+report_main_step () {
+  echo ""
+  echo "-------------------------------------------------"
+  echo "MAIN STEP: $1"
```

Comment:
The idiomatic buildkite way is `echo "--- $1"`.  It'll log a collapsible group of messages.  This is fine too if it works for you.